lib/bundler/source/rubygems/remote.rb

```diff
@@ -25,8 +25,7 @@ def cache_slug
 
             cache_uri = original_uri || uri
 
-            # URI::File of Ruby 2.6 returns empty string when given "file://".
-            host = defined?(URI::File) && cache_uri.is_a?(URI::File) ? nil : cache_uri.host
+            host = cache_uri.to_s.start_with?("file://") ? nil : cache_uri.host
```

Comment:
What is this change? Just a different way to achieve the same thing?

Reply:
Very good question.

My idea is to fix the issue that when `file://` URI is used with "localhost" (`file://localhost/path` instead of `file:///path`), then `localhost` would be prepended to the cache path, breaking [this spec](https://travis-ci.org/bundler/bundler/jobs/528808519).

Unfortunately I still need to use "localhost-type" file URIs for older rubies, because otherwise I can't seem to get a consistent behavior across versions (I guess because of bugs fixed in recent rubies):

https://github.com/bundler/bundler/blob/cdcde3e3e5d064dbcad7d20570de1732ba68ef9b/spec/support/path.rb#L69

The previous code use `file:/`style URI (single colon) to bypass this and get the spec passing:

https://github.com/bundler/bundler/blob/3d94636113db3f373be05514b7ae982db26b42e2/spec/install/global_cache_spec.rb#L197

But those don't work well on Windows.

The change makes sense to me since it makes the behavior consistent across ruby versions: we never include the host for `file://` URIs.